_rules/text-contrast-enhanced-09o5cg.md

```diff
@@ -0,0 +1,412 @@
+---
+id: 09o5cg
+name: Text has enhanced contrast
+rule_type: atomic
+description: |
+  This rule checks that the highest possible contrast of every text character with its background meets the enhanced contrast requirement.
+accessibility_requirements:
+  wcag20:1.4.6: # Contrast (Enhanced)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+  wcag-technique:G17: # Ensuring that a contrast ratio of at least 7:1 exists between text (and images of text) and background behind the text
+    forConformance: false
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+  wcag-technique:G18: # Ensuring that a contrast ratio of at least 4.5:1 exists between text (and images of text) and background behind the text
+    forConformance: false
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - Accessibility Tree
+  - DOM Tree
+  - CSS Styling
+  - Language
+acknowledgments:
+  authors:
+    - Jean-Yves Moyen
+---
+
+## Applicability
+
+This rule applies to any [visible][] character in a [text node][] that is a [child][] in the [flat tree][] of an HTML element, except if the [text node][] has an [ancestor][] in the [flat tree][] for which one of the following is true:
+
+- **widget**: the ancestor has a [semantic role][] that inherits from `widget`; or
+- **disabled label**: the ancestor is used in the [accessible name][] of a `widget` that is [disabled][]; or
```

Comment:
```suggestion
- **disabled label**: the ancestor is used in the [accessible name][] of a **widget** that is [disabled][]; or
```

I believe the "widget" in this sentence is referring to the widget condition that was expressed in the previous sentence. If it is referring to the "widget" semantic role then it needs to be rephrased because semantic roles do not have accessible names (elements do).

Reply:
🤔 Actually wondering if it's worth creating a definition of "semantic widget", the same way we have "semantic link" (in several PRs now).
WDYT?

Reply:
I think it is worth it 👍